### PR TITLE
infra container: replace pause with catatonit

### PR DESCRIPTION
**Pull request by vrothberg**
Podman has been using catatonit for a number of years already.
Thanks to @giuseppe, catatonit is now able to run as a pause
process which allows us to replace the pause binary entirely.

[NO NEW TESTS NEEDED] since no functionality has been added.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Review by Luap99 — COMMENTED** [review]
LGTM
We have to highlight this in the release notes to make sure catatonit is a hard dependency in all distros.

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by vrothberg** [comment]
Probably need to update the VMs and images to include catatonit 0.1.7. @cevich WDYT?

**Comment by rhatdan** [comment]
I updated fedora to use newer catatonit.

**Comment by vrothberg** [comment]
Thanks, @cevich.  I think we can also rerun `hack/install_catatonit.sh` on the images

**Comment by cevich** [comment]
>  I think we can also rerun hack/install_catatonit.sh on the images

Yep, that's exactly what we do.  So I'll just bump it up in podman, and the images will cache it for us as I build them.

**Comment by vrothberg** [comment]
@cevich do we need to change the images or shall we force-install catatonit?

File: pkg/specgen/generate/pod_create.go
Comment (edsantiago) [inline pkg/specgen/generate/pod_create.go:35]: I don't see how this can work unless `containers.conf` is updated to include `/usr/libexec/catatonit` in `helper_binaries_dir`. Is that in progress (presumably elsewhere)?

Comment (Luap99) [reply pkg/specgen/generate/pod_create.go:35]: It should work, at least on fedora `/usr/libexec/podman/catatonit` is a symlink to the actual catatonit path.
```
$ ll /usr/libexec/podman/catatonit 
lrwxrwxrwx. 1 root root 32 Sep 16 15:40 /usr/libexec/podman/catatonit -> /usr/libexec/catatonit/catatonit
```

Comment (edsantiago) [reply pkg/specgen/generate/pod_create.go:35]: Oops - thank you, I had missed that.

**Comment by cevich** [comment]
> do we need to change the images or shall we force-install catatonit?

Nope, no new images needed - they only cache catatonit based on podman main-branch.  So forcing it (as my PR did) is sufficient.  It'll just run a few seconds slower until new VM images pick up the change and cache it for us.

**Comment by vrothberg** [comment]
> > do we need to change the images or shall we force-install catatonit?
> 
> Nope, no new images needed - they only cache catatonit based on podman main-branch. So forcing it (as my PR did) is sufficient. It'll just run a few seconds slower until new VM images pick up the change and cache it for us.

The script will only install catatonit if it's not already present on the host.  Bumping the version will not automatically reinstall it if executed.  Or am I missing something?

**Comment by cevich** [comment]
> The script will only install catatonit if it's not already present on the host.

Oh crud, good catch!  So we do need the new images then (I just started a build this morning).  I believe I have all the testing-problems worked out for F35, but new ones could creep in at any moment :disappointed:  So if that workflow ends up being "too slow" for this PR, I can work on adding a '--force' option to the installer script - since that will be more reliable than trying to do version-detection/conditional install.

**Comment by vrothberg** [comment]
@cevich, thanks for confirming.  I can do the `--force` here and once we have new images, we can drop that.

**Comment by cevich** [comment]
Another possibility is https://github.com/containers/podman/pull/12256 (freshly built yesterday) which could go in before the F35 update (assuming tests pass).

File: .cirrus.yml
Comment (cevich) [inline .cirrus.yml:98]: Perfect.

**Comment by edsantiago** [comment]
@vrothberg test failures look real: I restarted both of them once (flake check) but both failed again, same way.

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming the tests get happy

**Comment by vrothberg** [comment]
It doesn't seem to install catatonit yet:
```
[+0002s] ./hack/install_catatonit.sh
[+0002s] skipping ... catatonit is already installed
```

**Comment by vrothberg** [comment]
Trying an alternative via contrib/cirrus/setup_environment.sh

**Comment by vrothberg** [comment]
APIv2 fail is legit. Fixed it already but I want to wait for e2e and system tests before pushing.

**Comment by cevich** [comment]
> It doesn't seem to install catatonit yet:

Perhaps try rebasing.  The 'main' branch has fresh images, which should (I haven't checked) include the updated/cached catatonit.

**Comment by vrothberg** [comment]
> LGTM We have to highlight this in the release notes to make sure catatonit is a hard dependency in all distros.

Heads up for packages: catatonit is now a hard-requirement for running Pods.
@lsm5 @jnovy @siretart

**Comment by cevich** [comment]
...confirmed, 0.1.7 is in the current VM images.

**Comment by rhatdan** [comment]
The spec file should be updated to require it.

**Comment by vrothberg** [comment]
> The spec file should be updated to require it.

We have a chicken-egg problem or is catatonit 0.17.0 already available in F3{3,4,5}?

**Comment by vrothberg** [comment]
> The spec file should be updated to require it.

Ah, already done in this PR.  I will drop the force-install of catatonit now and hope VMs are good.

**Comment by rhatdan** [comment]
BTW F33 support should be dropped now.

**Comment by siretart** [comment]
Thanks for the heads up,  turns out we already have it in Debian 😀

**Comment by vrothberg** [comment]
@cevich, it doesn't seem to be in the Ubuntu 2110 images.

**Comment by vrothberg** [comment]
Went back to force-installing to unblock the PR.

**Comment by vrothberg** [comment]
@containers/podman-maintainers lets get this in

File: contrib/cirrus/setup_environment.sh
Comment (edsantiago) [inline contrib/cirrus/setup_environment.sh:27]: What will this mean for bodhi gating tests? I see catatonit-0.1.6-1 on f34 and f35. Will 0.17.0 be packaged for those, or will this podman version never exist for f34/f35?

Comment (vrothberg) [reply contrib/cirrus/setup_environment.sh:27]: It's currently in testing for 34/35 -> https://bodhi.fedoraproject.org/updates/?search=catatonit

Comment (edsantiago) [reply contrib/cirrus/setup_environment.sh:27]: ack, thanks. Guess I better go play with it & give it karma.

**Comment by edsantiago** [comment]
I built this on f34 and f35. Tested using `catatonit-0.1.6-1`, tests fail. Then `0.1.7-1`, tests pass. This is not an LGTM, I haven't done an actual code review; it's just a super-basic sanity check.

**Comment by cevich** [comment]
> Went back to force-installing to unblock the PR.

Hmmm, pretty sure I remember grabbing a VM and checking it was there and `--version` said 0.1.7.  I think I checked the F35 image only though.  

```
$ podman run -it --rm quay.io/libpod/fedora_podman:c4955591916388352 /usr/libexec/catatonit/catatonit --version
tini version 0.1.7_catatonit
```

```
podman run -it --rm quay.io/libpod/ubuntu_podman:c4955591916388352
root@a94ec970614f:/var/tmp/go/src/github.com/containers/podman# type -p catatonit
/usr/bin/catatonit
root@a94ec970614f:/var/tmp/go/src/github.com/containers/podman# catatonit --version
tini version 0.1.5_catatonit
```

Aww nuts.  I think in Ubuntu it must get installed from a package, then the hack script finds it and doesn't install the newer one.  We don't mention the package by name during image build, so it must get pulled in by dependency :cry:

**Comment by cevich** [comment]
Update: Yep, [found it (repo)](https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_21.10/).  @lsm5 - something in kubik testing has a dependency on catatonit 0.1.5, could you update that and the package to 0.1.7?

**Comment by cevich** [comment]
@vrothberg a `--force` install from the script (to `/usr/libexec/...`) could be dangerous given the packaged version is installed in `/usr/bin/`.  That means the tests or whatever is running in the VM is a `$PATH` sequence|presence good away from using the wrong binary :disappointed: 

Let's see if @lsm5 can turn around a package update quickly, then I can rebuild the VM images (doing it for 'main' branch should be mostly painless).

**Comment by edsantiago** [comment]
/lgtm
/hold

I'm totally OK with this approach, just to get this PR in and give it testing time. There are plenty of update-the-VMs PRs in flight and upcoming, we can remove the install-catatonit hack soon.

**Pull request merged**
